### PR TITLE
Add dry-run option to flagged forum post digest

### PR DESCRIPTION
**Pull request by stvstnfrd**
A new "--dry-run" flag has been added, which will print the course_id
and email address for intended recipients, instead of sending an actual
email.

Other, minor changes have been included, in preparation of deploying
this feature.

### CODE REVIEW

**Comment by stvstnfrd** [comment]
Related configuration changes at
https://github.com/Stanford-Online/configuration/pull/2

**Comment by stvstnfrd** [comment]
@jbau @jrbl to complete the circle...

File: notifier/management/commands/forums_digest_flagged.py [hunk 1]
```diff
@@ -14,12 +14,22 @@ class Command(BaseCommand):
...
+            help='send digests for the specified courses only' +
+                ' (defaults to fetching course list via Heroku)',
+        ),
+        make_option(
+            '--is-dry-run',
+            action='store_true',
+            dest='is_dry_run',
+            default=False,
+            help='display the course_id and email address' +
+                ' for each recipient',
```
Comment (jrbl) [inline notifier/management/commands/forums_digest_flagged.py]: I'd add a clause indicating that no email is actually sent.

File: notifier/tasks.py [hunk 2]
```diff
@@ -157,7 +157,7 @@ def _time_slice(minutes, now=None):
     return (dt_start, dt_end)
 
 @celery.task(max_retries=settings.DAILY_TASK_MAX_RETRIES, default_retry_delay=settings.DAILY_TASK_RETRY_DELAY)
-def do_forums_digests_flagged(input_file=None):
+def do_forums_digests_flagged(input_file, is_dry_run):
```
Comment (jrbl) [inline notifier/tasks.py:160]: I'm fuzzy on how is_dry_run makes it from do_forums_digests_flagged() to batch(). Could you connect the dots for me?

File: notifier/management/commands/forums_digest_flagged.py (hunk 1 above)
Comment (stvstnfrd) [inline notifier/management/commands/forums_digest_flagged.py]: Added at 6d7a5c3

File: notifier/tasks.py [hunk 3]
```diff
@@ -270,14 +279,14 @@ def batch(messages):
         for message in messages:
             batch.append(message)
```
Comment (jrbl) [inline notifier/tasks.py:283]: I know you didn't write it, but having a local variable with the name of its parent context makes me sad.

Comment (jrbl) [inline notifier/tasks.py:283]: I'm not actually asking you to do anything about it though, unless you are saddened too.

File: notifier/tasks.py [hunk 4]
```diff
@@ -256,7 +256,16 @@ def get_messages(input_file):
                     'recipient': user,
                 }
 
-    def batch(messages):
+    def send(batch, is_dry_run):
+        if is_dry_run:
+            for message in batch:
+                email = message.get('recipient', {}).get('email')
```
Comment (jrbl) [inline notifier/tasks.py]: Could you call this like email_to just so it's immediately obvious to the most casual observer that it's an address and not the contents of a message?

**Comment by jrbl** [comment]
This looks good to me. I think it'll be pretty useful.

File: notifier/tasks.py (hunk 2 above)
Comment (stvstnfrd) [inline notifier/tasks.py:160]:
``` python
def do_forums_digests_flagged(input_file, is_dry_run):
    # ...
    batch(messages, is_dry_run)

    def batch(messages, is_dry_run):
        # ...
        send(batch, is_dry_run)

    def send(batch, is_dry_run):
        if is_dry_run:
            for message in batch:
                email = message.get('recipient', {}).get('email')
                course_id = message.get('course_id')
                print("{course_id}  {email}".format(course_id=course_id, email=email))
        else:
            generate_and_send_digests_flagged.delay(batch)

```

Comment (stvstnfrd) [inline notifier/tasks.py:160]: `do_forums_digests_flagged` actually only makes 2 function calls, after defining a few helper functions:

```
    messages = get_messages(input_file)
    batch(messages, is_dry_run)

```

File: notifier/tasks.py (hunk 4 above)
Comment (stvstnfrd) [inline notifier/tasks.py]: Renamed at 01f08f9

File: notifier/tasks.py (hunk 2 above)
Comment (jrbl) [inline notifier/tasks.py:160]: Oh! Thanks. In the git diff view it wasn't at all clear that those were inner functions of the do_forums_digests_flagged. Now I think I'm squared away.

File: notifier/tasks.py (hunk 3 above)
Comment (stvstnfrd) [inline notifier/tasks.py:283]: 1. Valid point.
2. I did, regrettably, write this.
3. Renamed batch -> send_in_batches at 5183e52
4. `:-(      ----->      :-)`

**Comment by jrbl** [comment]
Yay, :+1: from me. Remember to squash before merge.

**Comment by stvstnfrd** [comment]
@kluo Did you pull these into your PR?

**Comment by kluo** [comment]
@stvstnfrd Yea I pulled the 5 commits from here and then worked from there

**Comment by stvstnfrd** [comment]
Cool.

**Pull request closed without merging**